src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,288 @@
+import type { ReactInstance } from 'ReactInstanceType';
+
+type ReactTestRendererJSON = {
+  type: string,
+  props: { [propName: string]: string },
+  children: null | Array<string | ReactTestRendererJSON>,
+  $$typeof?: any
+}
+
+let instanceCounter = 0;
+
+class TestContainer {
+  constructor(rootID, createNodeMock) {
+    this.rootID = rootID;
+    this.createNodeMock = createNodeMock;
+  }
+  appendChild(child) {}
+  insertBefore(beforeChild, child) {}
+  removeChild(child) {}
+  toJSON() {}
+}
+
+class TestComponent {
+  constructor(type, props, rootContainerInstance) {
+    this.id = instanceCounter++;
+    this.type = type;
+    this.props = props;
+    this.children = [];
+    this.rootContainerInstance = rootContainerInstance;
+
+    Object.defineProperty(this, 'id', { value: this.id, enumerable: false });
+    Object.defineProperty(this, '$$typeof', {
+      value: Symbol.for('react.test.json'),
+    });
+  }
+
+  appendChild(child) {
+    const index = this.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    this.children.push(child);
+  }
+
+  removeChild(child) {
+    const index = this.children.indexOf(child);
+    if (index === -1) {
+      throw new Error('This child does not exist.');
+    }
+    this.children.splice(index, 1);
+  }
+
+  insertBefore(beforeChild, child) {
+    const index = this.children.indexOf(child);
+    if (index !== -1) {
+      this.children.splice(index, 1);
+    }
+    const beforeIndex = this.children.indexOf(beforeChild);
+    if (beforeIndex === -1) {
+      throw new Error('This child does not exist.');
+    }
+    this.children.splice(beforeIndex, 0, child);
+  }
+
+  update(type, props) {
+    // console.log('update', type, props, this.children);
+    // need to manually update text nodes
+    this.type = type;
+    this.props = props;
+  }
+
+  toJSON() {
+    // eslint-disable ignore the children
+    const {children, ...props} = this.props;
+    // eslint-enable
+    const json = {
+      type: this.type,
+      props: props,
+      children: null,
+    };
+
+    Object.defineProperty(json, '$$typeof', {value: Symbol.for('react.test.json')});
+    if (typeof children === 'string') {
+      json.children = [children];
+    } else {
+      var childrenJSON = [];
+      this.children.forEach((child) => {
+        if (typeof child.toJSON === 'function') {
+          childrenJSON.push(child.toJSON());
+        } else if (typeof child.text !== 'undefined') {
+          childrenJSON.push(isNaN(+child.text) ? child.text : +child.text);
+        }
+      });
+      json.children = childrenJSON.length ? childrenJSON : null;
+    }
+    return json;
+  }
+}
+
+type Instance = TestComponent;
+
+var TestRenderer = ReactFiberReconciler({
+  getRootHostContext(rootContainerInstance : Container) : HostContext {
+    return {};
+  },
+
+  getChildHostContext(
+    parentHostContext : HostContext,
+    type : string,
+  ) : HostContext {
+    return {};
```

Comment:
Please return the same object (emptyObject like in ReactNoop) so it doesn't push stack unnecessarily.